src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -53,34 +55,62 @@ def acr_task_create(cmd,  # pylint: disable=too-many-locals
                     resource_group_name=None,
                     target=None,
                     auth_mode=None):
+
+    registry, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    if context_path == NULL_CONTEXT:
+        context_path = None
+        commit_trigger_enabled = False
+        pull_request_trigger_enabled = False
+
     if (commit_trigger_enabled or pull_request_trigger_enabled) and not git_access_token:
         raise CLIError("If source control trigger is enabled [--commit-trigger-enabled] or "
                        "[--pull-request-trigger-enabled] --git-access-token must be provided.")
 
-    if file.endswith(ALLOWED_TASK_FILE_TYPES):
-        FileTaskStep = cmd.get_models('FileTaskStep')
-        step = FileTaskStep(
-            task_file_path=file,
-            values_file_path=values,
+    if cmd_value:
+        YAML_TEMPLATE = "steps: \n  - cmd: {{ .Values.image }}\n"
+        values_content = "image: {0}\n".format(cmd_value)
+
+    if file == "-":
+        import sys
+        YAML_TEMPLATE = ""
+        for s in sys.stdin.readlines():
+            YAML_TEMPLATE += s
+        values_content = ""
+
+    if values_content is not None:
+        import base64
+        EncodedTaskStep = cmd.get_models('EncodedTaskStep')
+        step = EncodedTaskStep(
+            encoded_task_content=base64.b64encode(YAML_TEMPLATE.encode()).decode(),
+            encoded_values_content=base64.b64encode(values_content.encode()).decode(),
             context_path=context_path,
             context_access_token=git_access_token,
```

Comment:
Since context_path is NULL in this case, you probably don't need to pass context_access_token.

Reply:
The default value of access_token is None.